examples/embeddable_examples/public/todo/todo_embeddable_factory.tsx

```diff
@@ -47,11 +51,13 @@ interface StartServices {
   openModal: OverlayStart['openModal'];
 }
 
-export class TodoEmbeddableFactory
+export class TodoEmbeddableFactory extends BindEmbeddableFactoryDefinitionOutputType<TodoOutput>
```

Comment:
because `output` is not used real output type wasn't inferred. 
I came up with this artificial hack to bind `output` type to definition and to factory throw definition. 
Not sure if this is needed or if there is a better way

Reply:
Interesting, I may need you to explain to me over zoom how this works or what this does.

Reply:
@stacey-gammon: tried to explain: [tsplayground](https://www.typescriptlang.org/play/#code/PTAEHUEkBUAkHkCq1RwKKlgQQMIGkAuAKBIFMAPABwHsAnAF1HoE9LTRIA7AZ3oENOAY1LRWpAGK1qAW3F9B9Os1FsAPEVCg5CpStJEAfKAC8W+YtrKxoCvVKcAJtzM7LqgJacAZqVocANKCePn7wBhqgAPwcPPxCpKqQgWERBKCcpABuvgDcJJ52tF7y7Fy8AsKJnJQAroymAN4Avsl1tfWgzUYNEZ7taVzteZrUbXVp8GP0eU35nIXFwi4WzFXtJp0toJP0641N3RGCtKR8dgAUnu707nwANoPjMe0AlGmZ1O4OM3MLJajUBzUR6MHqaRRAtK8WieADmPwKvkW7GggOoO3WYKYaKh9BhnHhRFmgjufG4zlRQO0KyC0kod1I0ns9Gc1KUqkpwOqdUCnIxdUOmmOpwuVxu9xBaU5IJenQimgis1mJBYbABQLKcWEeg2moqIjEkhkbMseg5aJNzHCRBA6WoAHdQIIABakQQAa1A9udZyCFLReviegAdEFGO5uANYvrzRrufRAjVOO7OA7OEZALwbgEo9m1geg+8POABGbr4NW47FGuzqfrtjBO9zuzFA5dIDlA1E46uolpIudAAGVqExXaAfR6OC2K+2+Ax3F53IJbncm6A+A4nB2ph2u4saYpQEXPDOmGJiLaoHBULAMNh8MQSCSyc4AELH-n0dSaD8bZqGOWaAA+oBR6OB+eiwPI7rRKY5xJg4pALhkM7OEmKZprKZLbFMOSgLa+YRh2RYAFZuowgijHc7YlmunDMARBKBO4dJ0HEjD5r69rsK27YAAYfrxp5qq6JxEkQqrsJa4juOQbZrDW+ytNWHRdBslryQm2HKUYABkoBvmBUyqB+Bh5BQNAMEJpTRkGhpSLI5hKNJskOGaESWno-6mB51i2PYm5STJcnBL4ARBN4oUpJo0SBpUSTbOEmhpBk2S0Hkj6kuS3aBS5NjkHYjivu+Rl8lMRjMfSjLMqyjmWM5cnSvGvJoiZAFOg2oqcNcy6St2MptQqmhKiq1iNeU8T1a51imLFBpsEaDmuMwk1mpyOVttatoZoAMnt9TZwgrdYhGzbGXLtM1QImUQQA)

I don't like it and likely will remove from this pr if we won't find anything better. 

And this is what it allowed to remove: 
https://github.com/elastic/kibana/pull/67783/files#diff-b8db5a716bad0cda3b5fc2c730624b7dL87
no need for explicit `<TodoRefOutput>` casting there

Reply:
Haven't looked in this more yet, will follow-up

Reply:
Better reproduction: [link](https://www.typescriptlang.org/play/#code/PTAEHUEkBUAkHkCq1RwKKlgQQMIGkAuAKBIFMAPABwHsAnAF1HoE9LTQAxAQwGN67m0VqQ61qAWwAipAGYBLAHZz6c6gqFsAPEVChpMogD5QAXj2zQFeqQUATAM7n5SlWs2KZpWqEgAaUB5eoPD+gd5ohjqgAPycvPy0zJp+IRFRBKAKpABuXgDcJESK1rQyvOyQaOIARqS2tlzVADakyf7wxgDeUQDmpPSQCpQArvQAFACUGZAFun308KMj41PBBQC+JI329LTxoDxNXPaOAELHpFW19Y0tbcHGcuKULeI29I6VNXUNza1+D1A3Tm-UGy0m0yBUV0tH6w1oCiYAAs5PYAHSKZazUCbEELJajCHBKG6GFwhHI1Fo6gE+jYzZRHhqHa0YZ8OhjShiax8OoBIajab+LnUHnWWygGn0ZYZeATKFRTYM4peMo8djcdmJZICxhmTrrdq00xAw2gK4-W7sKw2Bw+C03P465a+RbS0bGMxfa6-O5g0ZG930QxdRmwrjWMaKZRyLhNf30aa61YO33sAA+oGGdlkijq2PsEn6KIUPTQTXspCJ1Wo1BaXAUGxILDYTmjrgUztGJoNgeWPbNqatlnI1jsnyHToTrtpnvt30dft1fY9c4ACnIeABrTSahJJafBWn+SctQz+ADkPHD1gvkSKChKavY0GotmoCZJTDf1AyLMUPRNiqpTlKgP5uv2wLfu+f67ABTYkIcxyOK+76ntao62mcFzoZoqEfsuYHvhBHpfkyCgsmyCRRrqGT4Qm8pQboMjZnwqiIvMJErHR4HGkxpKwvQ8KIpiozQji4n2MMbC0DRLqgJxtKTBMipEAySEnER1B7gI+jtuxATPK87yOHpLjsXhP6HvhXGhro16kBGVb6XGCY8e+DFfmSQkUlkADuWnoWMnT8DBoAXkipBNE01AXusEz0mphQthq8QCBo7BmDpiQZaIEhmTGagZZZ77ZcwBUdveICZNQAU8JF26gH5SIRgEjhlRlaIBIwqIZGVJUEfJ2ZbgotUKP4+HocYgC8G4AlHtEEAA)